### ISSUE TITLE
Removal of <script> functionality needs to be added to https://reactjs.org/docs/ and a warning displayed while in development mode.

### ISSUE THREAD

**Issue by infoeon** [facebook/react#19865]
When a <script> tag is used within JSX there is functionality that tricks the browser into not executing the contained code or from fetching the `src`. There is a code comment indicating this behaviour:

`Create the script via .innerHTML so its "parser-inserted" flag is set to true and it does not execute`

but for those who are expecting the tag to work like _every other html tag_ in React it is a bit of a surprise that wastes several hours tracking down the cause.  This "feature" is obviously there for security reasons, but there are numerous ways around it (changing the case of a letter such as <sCript src="...">).

Please either remove this feature and add a warning while in dev mode stating that adding <script> tags is not best practices, or keep the feature but warn that <script> tags are not supported inline and document this behaviour in the React docs. I personally recommend the first once since it simplifies the React code base while allowing the developer to still proceed at their own risk.  The "risk" is minor since this isn't readily exploitable as the dangerouslySetHtml feature.

**Comment by gaearon** [comment]
Yeah I think it's pretty confusing since you still have it executing if you use server rendering. I agree it's worth revisiting.
